data/src/main/java/com/example/data/backendapi/dto/Customer.java

```diff
@@ -0,0 +1,63 @@
+package com.example.data.backendapi.dto;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+
+import java.util.Objects;
+
+/**
+ * CLient's objects with personal information
+ */
+@Entity
+public class Customer {
+
+    @Id
+    @GeneratedValue(strategy=GenerationType.AUTO)
+    private Long id;
```

Comment:
Consider to use `Lombok` annotation. That might significantly reduce boilerplate and make it easy to maintain. For caparison it could be just like that one:
```
@Entity
@Value
@NoArgsConstructor
@AllArgsConstructor
public class Customer {

    @Id
    @GeneratedValue(strategy=GenerationType.AUTO)
    Long id;
    String firstName;
    String lastName;
}
```

This is Data Object stereotype. So it must not contain the logic. Using Lombok here considered as an acceptable tradeoff.

Also you can take a look on new Java feature: `records`